recipes/beauty/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.43.0"
+
+
+class BeautyConan(ConanFile):
+    name = "beauty"
+    homepage = "https://github.com/dfleury2/beauty"
+    description = "HTTP Server above Boost.Beast"
+    topics = ("http", "server", "boost.beast")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+
+    requires = ("boost/[>1.70.0]@",
+                "openssl/1.1.1o@")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8",
+            "clang": "11",
+            "Visual Studio": "16",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"Compiler {self.name} must be at least {minimum_version}")
+        
+        if self.settings.compiler == "clang" and self.settings.compiler.libcxx != "libc++":
+            raise ConanInvalidConfiguration("Only libc++ is supported for clang")
+
+        if self.settings.compiler == "apple-clang" and self.options.shared:
+            raise ConanInvalidConfiguration("shared is not supported on apple-clang")
+
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("shared is not supported on Visual Studio")
+
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=self._source_subfolder)
+        cmake.build(target="beauty")
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        self.copy("version.hpp", dst=os.path.join("include", "beauty"), src=os.path.join(self.build_folder, "src", "beauty"))
+        self.copy("*.dll", src="bin", dst="bin", keep_path=False)
+        self.copy("*.lib", src="lib", dst="lib", keep_path=False)
+        self.copy("*.so*", src="lib", dst="lib", keep_path=False, symlinks=True)
+        self.copy("*.a", src="lib", dst="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.includedirs = ["include"]
```

Comment:
```suggestion
```
this is the default

Reply:
Ok, I will try when I remove this default. I do not have components this should be fine.